CMakeLists.txt

```diff
@@ -13,8 +13,10 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
 
-# TODO: re-enable /GL and /LTCG when we get a bigger build machine
-add_compile_options(/await /bigobj /GR- /permissive-)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    # TODO: re-enable /GL and /LTCG when we get a bigger build machine
+    add_compile_options(/await /bigobj /GR- /permissive-)
+endif()
```

Comment:
It wouldn't hurt to add an `else()` with `add_compile_options(-Werror)` for clang/gcc compile options.